### ISSUE TITLE
Bug: Fix tiny token amounts

### ISSUE THREAD

**Issue by murderteeth** [yearn/seafood#169]
When showing token amounts Seafood currently rounds out tokens that are very small against usd. For example Seafood shows 0.00 assets for this USDR vault when the correct value us 0.00000806.

https://seafood.yearn.watch/vault/0x531fDb1C8429F404116932226D487db94471C4b7
![image](https://github.com/yearn/seafood/assets/89237203/d5482c22-9603-4d4c-9638-8ceb3ad53b07)

Update Seafood to work like this instead:
![image](https://github.com/yearn/seafood/assets/89237203/c18ca6d6-1751-4309-aa23-2fd0d4edb94f)

### Tasks
- Update `formatNumber` to include compact forms for numbers less than zero using greek si units for milli, micro, nano, and pico:
https://github.com/yearn/seafood/blob/9d9645417d823fab231ffcb94fb0535a76b9d96d/src/utils/utils.js#L38

- Add tests here to demonstrate the new formats work:
https://github.com/yearn/seafood/blob/main/src/utils/utils.test.js

- Update the token component to support hover and click features. Use a localstorage value to store the user's current preference (unclicked vs clicked, ie compact vs expanded):
https://github.com/yearn/seafood/blob/9d9645417d823fab231ffcb94fb0535a76b9d96d/src/components/controls/Fields.tsx#L80

**Comment by heysourin** [comment]
Hi, I see here are two different tasks added inside one single issue.
1. Fix the decimals
2. Fix update and hover
Am I right?

**Comment by murderteeth** [comment]
Hi @heysourin! Yes, you are correct

**Comment by udhaykumarbala** [comment]
Hi @murderteeth, The formatNumber function already uses "m" to represent million. now the "milli" also has same character. can we change the million to "M"?

**Comment by murderteeth** [comment]
Good catch @udhaykumarbala! Yes, lets change the logic so that numbers greater than zero all use uppercase and all numbers less than zero use lowercase.

**Comment by Hari-Bombon** [comment]
Hello! I'm excited to start my journey in open source contribution as a beginner. Tech stack - reactjs , solidity, javascript
